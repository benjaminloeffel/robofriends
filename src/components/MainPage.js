import React, { useEffect } from "react";

import Head from "../components/Head";
import Header from "../components/Header";
import CardList from "../components/CardList";
import StatusBar from "../components/StatusBar";

const MainPage = ({
  searchTerm,
  users,
  isLoading,
  error,
  onSearchChange,
  onRequestUsers
}) => {
  useEffect(() => {
    onRequestUsers();
  }, [onRequestUsers]);

  const filterUsers = users => {
    return users.filter(user => {
      return user.name.toLowerCase().includes(searchTerm.toLowerCase());
    });
  };

  return (
    <div>
      <Head />
      <Header searchTerms={searchTerm} searchChange={onSearchChange} />
      {console.log()}
      <StatusBar
        loading={isLoading}
        error={error}
        resultLength={filterUsers(users).length}
      />
      {filterUsers ? <CardList users={filterUsers(users)} /> : null}
    </div>
  );
};

export default MainPage;
