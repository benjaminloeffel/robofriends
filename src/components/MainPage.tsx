import React, { useEffect } from "react";

import Head from "./Head";
import Header from "./Header";
import CardList from "./CardList";
import StatusBar from "./StatusBar";

import { User } from "../store/types";

type Props = {
  searchTerm: string;
  users: Array<object>;
  isLoading: boolean;
  error: Error;
  onSearchChange: () => void;
  onRequestUsers: () => void;
};

const MainPage: React.FC<Props> = ({
  searchTerm,
  users,
  isLoading,
  error,
  onSearchChange,
  onRequestUsers
}) => {
  useEffect(() => {
    onRequestUsers();
  }, [onRequestUsers]);

  const filterUsers = (users: any) => {
    return users.filter((user: User) => {
      return user.name.toLowerCase().includes(searchTerm.toLowerCase());
    });
  };

  return (
    <div>
      <Head />
      <Header searchTerms={searchTerm} searchChange={onSearchChange} />
      <StatusBar
        loading={isLoading}
        error={error}
        resultLength={filterUsers(users).length}
      />
      {filterUsers ? <CardList users={filterUsers(users)} /> : null}
    </div>
  );
};

export default MainPage;
