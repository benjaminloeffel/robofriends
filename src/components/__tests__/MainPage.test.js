import React from "react";
import "../../setupTests";
import { shallow } from "enzyme";
import MainPage from "../MainPage";

describe("MainPage", () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      searchTerm: "",
      users: [],
      isLoading: false,
      error: {},
      onSearchChange: jest.fn(),
      onRequestUsers: jest.fn(),
      filterUsers: 0
    };

    wrapper = shallow(<MainPage {...mockProps} />);
  });

  it("renders the <MainPage /> component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
