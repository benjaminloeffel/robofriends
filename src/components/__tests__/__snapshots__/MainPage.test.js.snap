// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders the <MainPage /> component 1`] = `ShallowWrapper {}`;
